### ISSUE TITLE
System hang on assigning tasks to units.

### ISSUE THREAD

**Issue by GrantSP** [LoneGazebo/Community-Patch-DLL#1747]
Bug Report Template

1. Mod Version (i.e Date - (4/23b)):
4/23b

2. Mod List (if using standard CPP set, leave blank):
+ FlagPromotions

3. Type of error (i.e. crash, interface bug, AI quirk): 
System hang.
Completely unresponsive to keyboard entry or mouse clicks, though the cursor does still move.
Even bringing up the TaskManager and trying to kill the exe that way is not possible, only a soft reboot will bring the system back inline.

4. Steps to reproduce:
The 3 or 4 times I have encountered it now I can say the only things in common was clicking with the mouse on an action in the unit action panel in the bottom left.
Different units, different tasks and different eras.

5. Additional information:
This is a dump from the most recent occurrence.
[CvMiniDump.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/253293/CvMiniDump.zip)

---------------------------

Supporting information:

Please note that you can attach .zip files to Issues by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files: Database.log and Lua.log needed
2. Minidump file (located in your Civ5.exe directory)
3. Screenshots (if needed)

----------------------------

**Comment by GrantSP** [comment]
Starting to see this again with the 5/21b release.
I can now say that the dump I provided previously was probably from another CTD, because the times I'm seeing this behaviour now no dumps are created.
Additionally it is also now happening when I press the assignment hotkey. I have purposely been avoiding using the unitpanel because of this issue.

Technically then this isn't a CTD because you aren't actually crashing to the desktop, the entire system is frozen with regard to mouse input and it is impossible to alt-tab to even get the task manager up and kill the process, you have to reset the system.

**Comment by GrantSP** [comment]
Just saw this again last night and I think it may be triggered because the unit was just gained a new promotion. I'll do some further tests today and see if I can trigger it the same way.

**Comment by GrantSP** [comment]
I'll close this as upon further occurrences I have determined that the system isn't freezing beyond return, it's just an inordinately long time while the game is processing in the background.

Just now saw this when I declared war upon a new civ in the previously unknown landmass to the west.
It happened around the time World Congress was allowed and apparently the only other surviving civ is a vassal of this new enemy. So I guess the game then had to determine all the alliance factors etc. and the system appeared to freeze. Not the same set of things as previously but nonetheless identical experience.
